contracts/contracts/multisig/Multisig.cairo

```diff
@@ -474,24 +436,113 @@ func execute_transaction{
     return (response_len=response.retdata_size, response=response.retdata)
 end
 
+# Sets number of required confirmations. The only way this can be invoked
+# is via a recursive call from execute_transaction -> set_confirmations_required.
+@external
+func set_confirmations_required{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    confirmations_required : felt
+):
+    const lower_bound = 1
+
+    let (caller) = get_caller_address()
+    let (contract_address) = get_contract_address()
+
+    with_attr error_message("caller shall be multisig"):
+        assert caller = contract_address
+    end
+
+    let (owners_len) = _owners_len.read()
+
+    with_attr error_message("invalid number of required confirmations"):
+        assert_in_range(confirmations_required, lower_bound, owners_len + 1)
+    end
+
+    _confirmations_required.write(confirmations_required)
+    ConfirmationsSet.emit(confirmations_required)
+    return ()
+end
+
+# Sets the owners field on the multisig. The only way this can be invoked
+# is via a recursive call from execute_transaction -> set_owners.
+# Number of required confirmations is decreased in case its larger than
+# number of new owners.
+@external
+func set_owners{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    owners_len : felt, owners : felt*
+):
+    alloc_locals
+
+    let (caller) = get_caller_address()
+    let (contract_address) = get_contract_address()
+
+    with_attr error_message("caller shall be multisig"):
+        assert caller = contract_address
+    end
+
+    with_attr error_message("invalid num of owners"):
+        assert_not_zero(owners_len)
+    end
```

Comment:
In the constructor we do currently allow this.

Reply:
Serum does check for presence of owners in constructor
 https://github.com/project-serum/multisig/blob/d6fbe734dc4026b9c8fdd2f9e78904112150fa74/programs/multisig/src/lib.rs#L40

I thinks that's part of this discussion.
https://github.com/smartcontractkit/starknet-multisig/pull/1#discussion_r885548586

Also it looks like that after deployment anyone could execute set_owners with any set of owners. So the first to send such message to Multisig would afterwards control it

Reply:
My only suggestion here is that these two flows:

- set_owners via constructor
- set_owners via public fn

..have different and inconsistent checks.

Which checks exactly we discussed in that other comment: [#1 (comment)](https://github.com/smartcontractkit/starknet-multisig/pull/1/files#:~:text=of%20this%20comment.-,%231%20(comment),-Also%20it%20looks)

But then please make those checks consistent for both flows. For example, constructor can just call `set_owners` internally and make sure it follows the same flow.